repository_service_tuf_api/config.py

```diff
@@ -28,14 +34,73 @@ class Config:
         }
 
 
+class PutResponse(BaseModel):
+    data: Dict[str, Any]
+    message: str
+
+    class Config:
+        with open("tests/data_examples/config/update_settings.json") as f:
+            content = f.read()
+        example_settings = json.loads(content)
+
+        schema_extra = {
+            "example": {
+                "message": "Settings successfully updated.",
+                "task_id": "06ee6db3cbab4b26be505352c2f2e2c3",
+            }
+        }
+
+
+class Settings(BaseModel):
+    # Only online roles can be dict keys
+    expiration: Dict[Roles.online_roles_values(), int]
+
+
+class PutPayload(BaseModel):
+    settings: Settings
+
+    class Config:
+        with open("tests/data_examples/config/update_settings.json") as f:
+            content = f.read()
+        example_settings = json.loads(content)
+
+        schema_extra = {
+            "example": example_settings,
+        }
+
+
+def put(payload: PutPayload):
+    if is_bootstrap_done() is False:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND,
+            detail={"error": "System has no repository metadata"},
+        )
+
+    task_id = get_task_id()
+
+    repository_metadata.apply_async(
+        kwargs={
+            "action": "update_settings",
+            "payload": payload.dict(by_alias=True, exclude_none=True),
+        },
+        task_id=task_id,
+        queue="metadata_repository",
+        acks_late=True,
+    )
+    data = {"task_id": task_id}
+
+    return PutResponse(data=data, message="Settings successfully updated.")
```

Comment:
```suggestion
    return PutResponse(data=data, message="Settings successfully submitted.")
```
Check also my comment about the status code.